### ISSUE TITLE
Progressbar crashed when terminal size change

### ISSUE THREAD

**Issue by ondra-m** [piotrmurach/tty-progressbar#13]
If progressbar is finished and terminal size is changed:

```
tty-progressbar-0.8.1/lib/tty/progressbar.rb:188:in `resize'
tty-progressbar-0.8.1/lib/tty/progressbar.rb:309:in `block in register_signals'
```

**Comment by peter-murach** [comment]
Thanks for reporting, will take a look.

**Comment by ondra-m** [comment]
Sorry, error was **Error: Cannot resize finished progress bar**.

**Comment by peter-murach** [comment]
Ah, probably it's better to log an error than crash.

**Comment by chrislewis60** [comment]
Hi Piotr, yes I've seen this a couple of times as well and would agree that it would be better if it didn't call 'fail' on line 188 (which stops everything including the process that it's showing progress for). I would think that just outputting/logging an error would be sufficient for most requirements. Thanks!

**Comment by piotrmurach** [comment]
Indeed Chris, it is definitely a bug scenario, I shall fix it soon.

**Comment by chrislewis60** [comment]
Super - thanks Piotr!

**Comment by ondra-m** [comment]
Would not be better to remove `register_signals` method?

**Comment by piotrmurach** [comment]
Hi Ondrej, agreed, I've decided to remove `register_signals` on the basis that exit codes and script behaviour should be left to developer to handle. Thanks.

**Comment by piotrmurach** [comment]
Released `v0.9.0`
